include/gsl/span

```diff
@@ -530,8 +532,38 @@ private:
     };
 
     storage_type<details::extent_type<Extent>> storage_;
+
+    // The rest is needed to remove unnecessary null check in subspans
+    constexpr span(pointer ptr, index_type count, bool basechecked) : storage_(ptr, count, basechecked) {}
+
+
+    template <bool Dependent = false,
```

Comment:
Maybe you could achieve the same effect here with slightly fewer characters by using a single boolean template parameter that tells `make_subspan` which case it is operating in. Then you wouldn't need the dummy `basechecked` parameter to distinguish them. Instead you could just use specialization to differentiate the two implementations.

Reply:
please see the updated changes.